src/apns_connection.erl

```diff
@@ -119,29 +120,38 @@ push_notification(ConnectionName, DeviceId, Notification, Headers) ->
 %%% gen_server callbacks
 %%%===================================================================
 
--spec init(connection()) -> {ok, State :: state()}.
-init(Connection) ->
+-spec init(list()) -> {ok, State :: state()}.
+init([Connection, Client]) ->
```

Comment:
In general, when you have something that has a fixed number of elements, you use a tuple (`{Connection, Client}`) not a list.